src/Extents.jl

```diff
@@ -177,44 +148,274 @@ Get the intersection of two extents as another `Extent`, e.g.
+When `strict=true`, any unshared dimensions cause the function to return `nothng`.
+"""
+
+const DE_9IM_DOC = """
+Conforms to the DE-9IM spatial predicates standard
+https://en.wikipedia.org/wiki/DE-9IM
+"""
+
+"""
+    contains(a::Extent, b::Extent; strict=false)
+
+`a` contains `b` if no points of `b` lie in the exterior of `a`, and 
+at least one point of the interior of `b` lies in the interior of `a`.
+If `b` has no interior points it is not contained in `a`.
+
+Identical to [`within`](@ref) with argument order reversed.
+
+$STRICT_DOC
+
+If there are no common dimensions, `false` is returned.
+
+$ORDER_DOC
+
+$DE_9IM_DOC
+"""
+contains(a::Extent, b::Extent; strict=false) = _do_bounds(all, _contain, a, b, strict)
+
+# Must contain interior points, not just boundary
+_contain(a::Tuple, b::Tuple) = _cover(a, b) && _hasinterior(b)
+
+"""
+    within(a::Extent, b::Extent; strict=false)
+
+`a` is within `b` if no points of `a` lie in the exterior of `b`, and 
+at least one point of the interior of `a` lies in the interior of `b`.
+If `a` has no interior points it is not contained in `b`.
+
+Identical to [`contains`](@ref) with argument order reversed.
+
+$STRICT_DOC
+
+If there are no common dimensions, `false` is returned.
+
+$ORDER_DOC
+
+$DE_9IM_DOC 
+"""
+within(a, b; kw...) = contains(b, a; kw...) # swapped order of `contains`
+
+"""
+    intersects(a::Extent, b::Extent; strict=false)
+
+`a` intersects `b` if `a` and `b` have at least one point in common
+(the inverse of [`disjoint`](@ref)).
+
+Returns `true` if the extents of all common dimensions share some values,
+including just the edge values of their range.
+
+$STRICT_DOC
+
+If there are no common dimensions with `strict=false`, `false` is returned.
+
+$ORDER_DOC
+
+$DE_9IM_DOC 
+"""
+intersects(a::Extent, b::Extent; strict=false) = _do_bounds(all, _intersect, a, b, strict)
+
+_intersect((min_a, max_a)::Tuple, (min_b, max_b)::Tuple) = 
+    (min_a <= min_b && max_a >= min_b) || (min_b <= min_a && max_b >= min_a)
+
+"""
+    distjoint(a::Extent, b::Extent; strict=false)
+
+`a` and `b` are disjoint if they have no point in common
+(the inverse of [`intersects`](@ref)).
+
+Returns `false` if the extents of all common dimensions share some values,
+including just the edge values of their range.
+
+$STRICT_DOC
+
+If there are no common dimensions when `strict=false`, `true` is returned.
+
+$ORDER_DOC
+
+$DE_9IM_DOC 
+"""
+function disjoint(a, b; kw...)
+    # negates `intersects`
+    x = intersects(a, b; kw...)
+    return isnothing(x) ? nothing : !x
+end
+
+"""
+    touches(a::Extent, b::Extent; strict=false)
+
+`a` and `b` have at least one point in common, but their interiors do not intersect. 
+
+Returns `true` if the extents of any common dimensions share boundaries.
+
+$STRICT_DOC
+
+If there are no common dimensions with `strict=false`, `false` is returned.
+
+$ORDER_DOC
+
+$DE_9IM_DOC 
+"""
+function touches(a::Extent, b::Extent; strict=false)
+    if intersects(a, b)
```

Comment:
Hmm, now a dimension could still intersect, while another dimension would touch. I think this needs to be either touch or not intersect at all.

Reply:
The intersecting part is to allow whole edges and sides to touch - if all edges intersect and one just touches then its a side or face touching. If all just touch then its only a corner.

Or do you mean related to `strict` ?

Reply:
(the rule is that no internal points are the same, but whole faces can be the same without that happening)

Reply:
Bump... would be good to merge this. 

Have you looked at the tests or the definition in https://en.wikipedia.org/wiki/DE-9IM

> "a touches b: they have at least one point in common, but their interiors do not intersect." 

I'm think that's what I have here, points/lines/sides touching is fine, but no overlap allowed.

Reply:
I think my point was that dim x and y (with y sharing interiors) could intersect, and only x would touch, which would pass your test here.

Reply:
Hmm, thinking about it some more, it seems that's a nice theoretic case that doesn't exist 😅 . Then this is good to go.

Reply:
Might be good to place a comment here for future reference why we need an intersect. Something like, just touching isn't enough (could be completely disjoint).

Also, maybe something like?

```
  | b
- + -
a |
```

Reply:
```
- +
  |
a + - 
  | 
- + b
  |
  + -
```

Reply:
Those graphocs are amazing we should put them in the docs. 

And I actually first implemented it exactly as you were suggesting but it didnt work 😂